### PR TITLE
[pyqgis] move qgis.core tricks to their own python file in additions folder

### PR DESCRIPTION
**Pull request by 3nids**
this makes the code much easier to read

### CODE REVIEW

**Review by nyalldawson — COMMENTED** [review]
Good idea!

Better put the copyright notice at the top of these files too

**Comment by nirvn** [comment]
@3nids , I think this (or a related sip commit) broke QGIS python support on my machine:

```
Couldn't load PyQGIS.
Python support will be disabled.


Traceback (most recent call last):
  File "", line 1, in 
  File "/home/webmaster/apps/share/qgis/python/qgis/__init__.py", line 72, in 
    from qgis.core import QgsFeature, QgsGeometry
  File "/home/webmaster/apps/share/qgis/python/qgis/core/__init__.py", line 29, in 
    from .additions.readwritecontextentercategory import ReadWriteContextEnterCategory
ModuleNotFoundError: No module named 'qgis.core.additions'


Python version:
3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0]

QGIS version:
3.1.0-Master 'Master', 395e864b68
```

**Comment by nyalldawson** [comment]
@3nids tried a full clean build?

**Comment by nirvn** [comment]
@nyalldawson , assuming that was directed to me, yes, I deleted my build directory and re-created it from scratch.

**Comment by nirvn** [comment]
Note: QGIS works fine when launched from the build's output directory; the error happens when I launch QGIS from the install directory (i.e. make install).

**Comment by nirvn** [comment]
Looking into {INSTALL_DIR}/share/qgis/python/qgis/core, all files which should be in the core/additions sub-directory are located in /core directory itself.

File: python/core/additions/qgsdefaultvalue.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+
+def _isValid(self):
```
Comment (luipir) [inline python/core/additions/qgsdefaultvalue.py]: what is the sense of this addition?

File: python/core/additions/qgsgeometry.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+
+def _geometryNonZero(self):
```
Comment (luipir) [inline python/core/additions/qgsgeometry.py]: is it an addition useful only for readability?

**Comment by tudorbarascu** [comment]
same error as @nirvn here, clean build clean install dir

**Comment by m-kuhn** [comment]
Interestingly also happens on the docker image `release-3_0`, has this been merged into 3.0 as well or is the master build image pushed to release-3_0?

https://travis-ci.org/opengisch/projectgenerator/jobs/379108133

**Comment by 3nids** [comment]
fix for the installed file on its way https://github.com/qgis/QGIS/pull/7000/files
looking at the docker now...

**Comment by 3nids** [comment]
@m-kuhn you're not using release_3-0 but latest as docker tag
image
https://travis-ci.org/opengisch/projectgenerator/jobs/379108133#L515-L519

File: python/core/additions/qgsdefaultvalue.py (hunk 1 above)
Comment (3nids) [reply python/core/additions/qgsdefaultvalue.py]: see in `qgis.core.__init__.py`, this is used for `QgsDefaultValue.__bool__` so you can do `if myValue: ....`

File: python/core/additions/qgsgeometry.py (hunk 2 above)
Comment (3nids) [reply python/core/additions/qgsgeometry.py]: see in `qgis.core.__init__.py`, this is used for `QgsGeometry.__bool__` so you can do `if myGeometry: ....`

**Comment by 3nids** [comment]
@tudorbarascu thanks, hopefully solved now

**Pull request merged**